### ISSUE TITLE
Buggy caching behaviour

### ISSUE THREAD

**Issue by sahanxdissanayake** [vercel/next.js#56915]
### Link to the code that reproduces this issue

https://github.com/sahanxdissanayake/refresh-issue/

### To Reproduce

1. Clone
2. Start the app
3. Go to http://localhost:3001/metrics and click the links

### Current vs. Expected behavior

Current
Everytime a link is clicked the data is fetched from the backend

Expected
Cached response(ISR)

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.1.0: Tue Sep 26 22:11:21 PDT 2023; root:xnu-10002.40.89.501.1~3/RELEASE_ARM64_T6000
Binaries:
  Node: 18.16.0
  npm: 9.5.1
  Yarn: N/A
  pnpm: 7.15.0
Relevant Packages:
  next: 13.5.6-canary.1
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.1.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router, Data fetching (gS(S)P, getInitialProps)

### Additional context

na

**Comment by sahanxdissanayake** [comment]
This is very strange. If I refresh the page the caching works, but after a few seconds the page is fetching again from the backend.

**Comment by ShawnAndrews** [comment]
Please try rolling back to v13.4.19. It is a known issue that ISR caching has been broken since v13.5.x

**Comment by balazsorban44** [comment]
@ShawnAndrews it's not known to us, do you have a reproduction to share? Would love to check.

_It's always recommended to upgrade to latest and report any bugs instead._

**Comment by sahanxdissanayake** [comment]
@balazsorban44 the code to recreate this bug should be attached in the issue template. 

1. Start the app
2. visit the link provided
3. Click around
4. Leave the browser tab for a couple of minutes
5. Come back, now everytime you click on a link the data is loaded from the backend again, but it shouldn't be ideally

**Comment by sahanxdissanayake** [comment]
maybe related https://github.com/vercel/next.js/issues/56232

**Comment by sahanxdissanayake** [comment]
@balazsorban44 is there anything I could do from my end to push forward with this issue?

**Comment by ShawnAndrews** [comment]
> @ShawnAndrews it's not known to us, do you have a reproduction to share? Would love to check.
> 
> _It's always recommended to upgrade to latest and report any bugs instead._

Here is probably the most notable thread about the issue but there are many: https://github.com/vercel/next.js/issues/56232
ISR and the "revalidate" command have been broken for over a month its kind of crazy for being such a core feature.

**Comment by ShawnAndrews** [comment]
I can confirm this issue was fixed in v14.0.2 per code commit https://github.com/vercel/next.js/pull/57737. This issue affected versions v13.4.19 to v14.0.2 (exclusive) which was a period of 2-months. It is completely unacceptable for such a core feature like ISR to be broken and ignored while major versions continue to be released. @sahanxdissanayake

**Comment by doinki** [comment]
Not resolved yet. The version I tested is 14.0.4 and when I use export const dynamic = 'force-dynamic' the route is not cached and continues to be requested.
